--- src/auth_des.c	2014-12-21 18:24:52.527242137 +0100
+++ src/auth_des.c	2014-12-21 18:24:54.387242113 +0100
@@ -38,7 +38,6 @@
 #include <string.h>
 #include <stdlib.h>
 #include <unistd.h>
-#include <sys/cdefs.h>
 #include <rpc/des_crypt.h>
 #include <syslog.h>
 #include <rpc/types.h>
@@ -52,7 +51,6 @@
 
 #if defined(LIBC_SCCS) && !defined(lint)
 #endif
-#include <sys/cdefs.h>
 
 #define USEC_PER_SEC		1000000
 #define RTIME_TIMEOUT		5	/* seconds to wait for sync */
--- src/auth_none.c	2014-12-21 18:24:52.527242137 +0100
+++ src/auth_none.c	2014-12-21 18:24:54.407242113 +0100
@@ -31,7 +31,6 @@
 static char *sccsid = "@(#)auth_none.c 1.19 87/08/11 Copyr 1984 Sun Micro";
 static char *sccsid = "@(#)auth_none.c	2.1 88/07/29 4.0 RPCSRC";
 #endif
-#include <sys/cdefs.h>
 __FBSDID("$FreeBSD: src/lib/libc/rpc/auth_none.c,v 1.12 2002/03/22 23:18:35 obrien Exp $");
 */
 
--- src/auth_time.c	2014-12-21 18:24:52.557242137 +0100
+++ src/auth_time.c	2014-12-21 18:24:54.537242111 +0100
@@ -25,7 +25,6 @@
  *	needed to deal with TCP connections.
  */
 
-#include <sys/cdefs.h>
 #include <stdio.h>
 #include <syslog.h>
 #include <string.h>
--- src/auth_unix.c	2014-12-21 18:24:52.527242137 +0100
+++ src/auth_unix.c	2014-12-21 18:24:54.317242114 +0100
@@ -26,7 +26,6 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 /*
  * auth_unix.c, Implements UNIX style authentication parameters.
--- src/authdes_prot.c	2014-12-21 18:24:52.547242137 +0100
+++ src/authdes_prot.c	2014-12-21 18:24:54.467242112 +0100
@@ -1,4 +1,3 @@
-#include <sys/cdefs.h>
 /*
  * Copyright (c) 2009, Sun Microsystems, Inc.
  * All rights reserved.
--- src/authunix_prot.c	2014-12-21 18:24:52.547242137 +0100
+++ src/authunix_prot.c	2014-12-21 18:24:54.507242111 +0100
@@ -26,7 +26,6 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 /*
  * authunix_prot.c
--- src/bindresvport.c	2014-12-21 18:24:52.557242137 +0100
+++ src/bindresvport.c	2014-12-21 18:24:54.567242111 +0100
@@ -26,7 +26,6 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 /*
  * Copyright (c) 1987 by Sun Microsystems, Inc.
--- src/clnt_bcast.c	2014-12-21 18:24:52.547242137 +0100
+++ src/clnt_bcast.c	2014-12-21 18:24:54.507242111 +0100
@@ -28,7 +28,6 @@
 /*
  * Copyright (c) 1986-1991 by Sun Microsystems Inc. 
  */
-#include <sys/cdefs.h>
 
 /*
  * clnt_bcast.c
--- src/clnt_perror.c	2014-12-21 18:24:52.527242137 +0100
+++ src/clnt_perror.c	2014-12-21 18:24:54.367242113 +0100
@@ -27,7 +27,6 @@
  */
 
 /*
-#include <sys/cdefs.h>
 */
 /*
  * clnt_perror.c
--- src/clnt_simple.c	2014-12-21 18:24:52.557242137 +0100
+++ src/clnt_simple.c	2014-12-21 18:24:54.547242111 +0100
@@ -29,7 +29,6 @@
  * Copyright (c) 1986-1991 by Sun Microsystems Inc. 
  */
 
-#include <sys/cdefs.h>
 
 /*
  * clnt_simple.c
--- src/crypt_client.c	2014-12-21 18:24:52.547242137 +0100
+++ src/crypt_client.c	2014-12-21 18:24:54.467242112 +0100
@@ -30,7 +30,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <err.h>
 #include <sys/types.h>
--- src/des_crypt.c	2014-12-21 18:24:52.557242137 +0100
+++ src/des_crypt.c	2014-12-21 18:24:54.547242111 +0100
@@ -39,7 +39,6 @@
 static char sccsid[] = "@(#)des_crypt.c	2.2 88/08/10 4.0 RPCSRC; from 1.13 88/02/08 SMI";
 #endif
 #endif
-#include <sys/cdefs.h>
 
 static int common_crypt( char *, char *, unsigned, unsigned, struct desparams * );
 int (*__des_crypt_LOCAL)() = 0;
--- src/getnetconfig.c	2014-12-21 18:24:52.527242137 +0100
+++ src/getnetconfig.c	2014-12-21 18:24:54.357242113 +0100
@@ -32,7 +32,6 @@
  
 #include <pthread.h>
 #include <reentrant.h>
-#include <sys/cdefs.h>
 #include <stdio.h>
 #include <errno.h>
 #include <netconfig.h>
--- src/getnetpath.c	2014-12-21 18:24:52.527242137 +0100
+++ src/getnetpath.c	2014-12-21 18:24:54.357242113 +0100
@@ -25,13 +25,11 @@
  * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
  * POSSIBILITY OF SUCH DAMAGE.
  */
-#include <sys/cdefs.h>
 
 /*
  * Copyright (c) 1989 by Sun Microsystems, Inc.
  */
 
-#include <sys/cdefs.h>
 #include <stdio.h>
 #include <errno.h>
 #include <netconfig.h>
--- src/getpeereid.c	2014-12-21 18:24:52.547242137 +0100
+++ src/getpeereid.c	2014-12-21 18:24:54.497242112 +0100
@@ -24,7 +24,6 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 #include <sys/param.h>
 #include <sys/socket.h>
--- src/getpublickey.c	2014-12-21 18:24:52.547242137 +0100
+++ src/getpublickey.c	2014-12-21 18:24:54.507242111 +0100
@@ -26,7 +26,6 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 /*
-#include <sys/cdefs.h>
 */
 
 /*
--- src/key_call.c	2014-12-21 18:24:52.547242137 +0100
+++ src/key_call.c	2014-12-21 18:24:54.447242112 +0100
@@ -30,7 +30,6 @@
  */
 
 
-#include <sys/cdefs.h>
 
 /*
  * key_call.c, Interface to keyserver
--- src/key_prot_xdr.c	2014-12-21 18:24:52.547242137 +0100
+++ src/key_prot_xdr.c	2014-12-21 18:24:54.447242112 +0100
@@ -33,7 +33,6 @@
  */
 /* Copyright (c)  1990, 1991 Sun Microsystems, Inc. */
 
-#include <sys/cdefs.h>
 
 /* 
  * Compiled from key_prot.x using rpcgen.
--- src/mt_misc.c	2014-12-21 18:24:52.527242137 +0100
+++ src/mt_misc.c	2014-12-21 18:24:54.287242114 +0100
@@ -1,5 +1,4 @@
 
-#include <sys/cdefs.h>
 #include <pthread.h>
 #include <reentrant.h>
 #include <rpc/rpc.h>
--- src/pmap_clnt.c	2014-12-21 18:24:52.527242137 +0100
+++ src/pmap_clnt.c	2014-12-21 18:24:54.317242114 +0100
@@ -26,7 +26,6 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 /*
  * pmap_clnt.c
--- src/pmap_getmaps.c	2014-12-21 18:24:52.527242137 +0100
+++ src/pmap_getmaps.c	2014-12-21 18:24:54.407242113 +0100
@@ -26,7 +26,6 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 /*
  * pmap_getmap.c
--- src/rpc_com.h	2014-12-21 18:24:52.527242137 +0100
+++ src/rpc_com.h	2014-12-21 18:24:54.267242114 +0100
@@ -40,7 +40,6 @@
 #ifndef _TIRPC_RPCCOM_H
 #define	_TIRPC_RPCCOM_H
 
-#include <sys/cdefs.h>
 
 /* #pragma ident	"@(#)rpc_com.h	1.11	93/07/05 SMI" */
 
--- src/rpc_generic.c	2014-12-21 18:24:52.527242137 +0100
+++ src/rpc_generic.c	2014-12-21 18:24:54.297242114 +0100
@@ -29,7 +29,6 @@
  * Copyright (c) 1986-1991 by Sun Microsystems Inc. 
  */
 
-#include <sys/cdefs.h>
 
 /*
  * rpc_generic.c, Miscl routines for RPC.
--- src/rpcb_st_xdr.c	2014-12-21 18:24:52.547242137 +0100
+++ src/rpcb_st_xdr.c	2014-12-21 18:24:54.417242113 +0100
@@ -35,7 +35,6 @@
  * routines used with the rpcbind stats facility.
  */
 
-#include <sys/cdefs.h>
 
 #include <rpc/rpc.h>
 
--- src/rpcdname.c	2014-12-21 18:24:52.547242137 +0100
+++ src/rpcdname.c	2014-12-21 18:24:54.447242112 +0100
@@ -25,7 +25,6 @@
  * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
  * POSSIBILITY OF SUCH DAMAGE.
  */
-#include <sys/cdefs.h>
 
 /*
  * rpcdname.c
--- src/svc_dg.c	2014-12-21 18:24:52.527242137 +0100
+++ src/svc_dg.c	2014-12-21 18:24:54.347242113 +0100
@@ -31,7 +31,6 @@
  * Copyright (c) 1986-1991 by Sun Microsystems Inc.
  */
 
-#include <sys/cdefs.h>
 
 /*
  * svc_dg.c, Server side for connectionless RPC.
--- src/svc_simple.c	2014-12-21 18:24:52.527242137 +0100
+++ src/svc_simple.c	2014-12-21 18:24:54.407242113 +0100
@@ -29,7 +29,6 @@
  * Copyright (c) 1986-1991 by Sun Microsystems Inc. 
  */
 
-#include <sys/cdefs.h>
 
 /*
  * svc_simple.c
--- src/svc_vc.c	2014-12-21 18:24:52.547242137 +0100
+++ src/svc_vc.c	2014-12-21 18:24:54.447242112 +0100
@@ -27,7 +27,6 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 /*
  * svc_vc.c, Server side for Connection Oriented based RPC. 
--- src/xdr.c	2014-12-21 18:24:52.527242137 +0100
+++ src/xdr.c	2014-12-21 18:24:54.267242114 +0100
@@ -26,7 +26,6 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 /*
  * xdr.c, Generic XDR routines implementation.
--- src/xdr_array.c	2014-12-21 18:24:52.527242137 +0100
+++ src/xdr_array.c	2014-12-21 18:24:54.387242113 +0100
@@ -27,7 +27,6 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 /*
  * xdr_array.c, Generic XDR routines impelmentation.
--- src/xdr_float.c	2014-12-21 18:24:52.547242137 +0100
+++ src/xdr_float.c	2014-12-21 18:24:54.507242111 +0100
@@ -27,7 +27,6 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 /*
  * xdr_float.c, Generic XDR routines implementation.
--- src/xdr_mem.c	2014-12-21 18:24:52.557242137 +0100
+++ src/xdr_mem.c	2014-12-21 18:24:54.567242111 +0100
@@ -26,7 +26,6 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 /*
  * xdr_mem.h, XDR implementation using memory buffers.
--- src/xdr_rec.c	2014-12-21 18:24:52.557242137 +0100
+++ src/xdr_rec.c	2014-12-21 18:24:54.567242111 +0100
@@ -27,8 +27,6 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
-#include <sys/cdefs.h>
 
 /*
  * xdr_rec.c, Implements TCP/IP based XDR streams with a "record marking"
--- src/xdr_reference.c	2014-12-21 18:24:52.547242137 +0100
+++ src/xdr_reference.c	2014-12-21 18:24:54.437242112 +0100
@@ -26,8 +26,6 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
-#include <sys/cdefs.h>
 
 /*
  * xdr_reference.c, Generic XDR routines impelmentation.
--- src/xdr_sizeof.c	2014-12-21 18:24:52.527242137 +0100
+++ src/xdr_sizeof.c	2014-12-21 18:24:54.297242114 +0100
@@ -34,7 +34,6 @@
  * when serialized using XDR.
  */
 
-#include <sys/cdefs.h>
 
 #include <rpc/types.h>
 #include <rpc/xdr.h>
--- src/xdr_stdio.c	2014-12-21 18:24:52.547242137 +0100
+++ src/xdr_stdio.c	2014-12-21 18:24:54.497242112 +0100
@@ -26,7 +26,6 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
 
 /*
  * xdr_stdio.c, XDR implementation on standard i/o file.
--- tirpc/rpc/auth.h	2014-12-21 18:24:52.567242136 +0100
+++ tirpc/rpc/auth.h	2014-12-21 18:24:54.697242109 +0100
@@ -48,7 +48,6 @@
 
 #include <rpc/xdr.h>
 #include <rpc/clnt_stat.h>
-#include <sys/cdefs.h>
 #include <sys/socket.h>
 #include <sys/types.h>
 
--- tirpc/rpc/auth_unix.h	2014-12-21 18:24:52.567242136 +0100
+++ tirpc/rpc/auth_unix.h	2014-12-21 18:24:54.627242110 +0100
@@ -45,7 +45,6 @@
 
 #ifndef _TIRPC_AUTH_UNIX_H
 #define _TIRPC_AUTH_UNIX_H
-#include <sys/cdefs.h>
 
 /* The machine name is part of a credential; it may not exceed 255 bytes */
 #define MAX_MACHINE_NAME 255
--- tirpc/rpc/clnt.h	2014-12-21 18:24:52.567242136 +0100
+++ tirpc/rpc/clnt.h	2014-12-21 18:24:54.627242110 +0100
@@ -42,7 +42,6 @@
 #include <rpc/clnt_stat.h>
 #include <rpc/auth.h>
 
-#include <sys/cdefs.h>
 #include <netconfig.h>
 #include <sys/un.h>
 
--- tirpc/rpc/clnt_soc.h	2014-12-21 18:24:52.567242136 +0100
+++ tirpc/rpc/clnt_soc.h	2014-12-21 18:24:54.657242109 +0100
@@ -46,7 +46,6 @@
  * with TS-RPC.
  */
 
-#include <sys/cdefs.h>
 
 #define UDPMSGSIZE      8800    /* rpc imposed limit on udp msg size */  
 
--- tirpc/rpc/des_crypt.h	2014-12-21 18:24:52.567242136 +0100
+++ tirpc/rpc/des_crypt.h	2014-12-21 18:24:54.717242109 +0100
@@ -43,7 +43,6 @@
 #ifndef _DES_DES_CRYPT_H
 #define _DES_DES_CRYPT_H
 
-#include <sys/cdefs.h>
 #include <rpc/rpc.h>
 
 #define DES_MAXDATA 8192	/* max bytes encrypted in one call */
--- tirpc/rpc/pmap_clnt.h	2014-12-21 18:24:52.567242136 +0100
+++ tirpc/rpc/pmap_clnt.h	2014-12-21 18:24:54.627242110 +0100
@@ -63,7 +63,6 @@
 
 #ifndef _RPC_PMAP_CLNT_H_
 #define _RPC_PMAP_CLNT_H_
-#include <sys/cdefs.h>
 
 __BEGIN_DECLS
 extern bool_t		pmap_set(u_long, u_long, int, int);
--- tirpc/rpc/pmap_prot.h	2014-12-21 18:24:52.567242136 +0100
+++ tirpc/rpc/pmap_prot.h	2014-12-21 18:24:54.687242109 +0100
@@ -71,7 +71,6 @@
 
 #ifndef _RPC_PMAP_PROT_H
 #define _RPC_PMAP_PROT_H
-#include <sys/cdefs.h>
 
 #define PMAPPORT		((u_short)111)
 #define PMAPPROG		((u_long)100000)
--- tirpc/rpc/pmap_rmt.h	2014-12-21 18:24:52.567242136 +0100
+++ tirpc/rpc/pmap_rmt.h	2014-12-21 18:24:54.687242109 +0100
@@ -41,7 +41,6 @@
 
 #ifndef _RPC_PMAP_RMT_H
 #define _RPC_PMAP_RMT_H
-#include <sys/cdefs.h>
 
 struct rmtcallargs {
 	u_long prog, vers, proc, arglen;
--- tirpc/rpc/rpc_com.h	2014-12-21 18:24:52.557242137 +0100
+++ tirpc/rpc/rpc_com.h	2014-12-21 18:24:54.607242110 +0100
@@ -41,7 +41,6 @@
 #ifndef _RPC_RPCCOM_H
 #define	_RPC_RPCCOM_H
 
-#include <sys/cdefs.h>
 
 /* #pragma ident	"@(#)rpc_com.h	1.11	93/07/05 SMI" */
 
--- tirpc/rpc/svc.h	2014-12-21 18:24:52.557242137 +0100
+++ tirpc/rpc/svc.h	2014-12-21 18:24:54.607242110 +0100
@@ -40,7 +40,6 @@
 
 #ifndef _TIRPC_SVC_H
 #define _TIRPC_SVC_H
-#include <sys/cdefs.h>
 
 /*
  * This interface must manage two items concerning remote procedure calling:
--- tirpc/rpc/svc_soc.h	2014-12-21 18:24:52.567242136 +0100
+++ tirpc/rpc/svc_soc.h	2014-12-21 18:24:54.647242110 +0100
@@ -38,7 +38,6 @@
 
 #ifndef _RPC_SVC_SOC_H
 #define _RPC_SVC_SOC_H
-#include <sys/cdefs.h>
 
 /* #pragma ident   "@(#)svc_soc.h  1.11    94/04/25 SMI" */
 /*      svc_soc.h 1.8 89/05/01 SMI      */
--- tirpc/rpc/xdr.h	2014-12-21 18:24:52.567242136 +0100
+++ tirpc/rpc/xdr.h	2014-12-21 18:24:54.647242110 +0100
@@ -40,7 +40,6 @@
 
 #ifndef _TIRPC_XDR_H
 #define _TIRPC_XDR_H
-#include <sys/cdefs.h>
 #include <stdio.h>
 #include <netinet/in.h>
 
