Index: m_whois.c
===================================================================
--- m_whois.c	(revision 133)
+++ m_whois.c	(working copy)
@@ -327,7 +327,7 @@
 			 * global oper or services.
 			 * -CodeM/Barubary
 			 */
-			if (MyConnect(acptr))
+			if (MyConnect(acptr) && (!IsOper(acptr) || IsOper(sptr) || IsServices(sptr)))
 				sendto_one(sptr, rpl_str(RPL_WHOISIDLE),
 				    me.name, parv[0], name,
 				    TStime() - acptr->last, acptr->firsttime);
