Index: m_names.c
===================================================================
--- a/src/modules/m_names.c	(revision 133)
+++ b/src/modules/m_names.c	(working copy)
@@ -163,9 +163,9 @@
 	for (cm = chptr->members; cm; cm = cm->next)
 	{
 		acptr = cm->cptr;
-		if (IsInvisible(acptr) && !member && !IsNetAdmin(sptr))
+		if (IsInvisible(acptr) && !member && !OPCanSeeSecret(sptr))
 			continue;
-		if (chptr->mode.mode & MODE_AUDITORIUM)
+		if (chptr->mode.mode & MODE_AUDITORIUM && !OPCanSeeSecret(sptr))
 			if (!is_chan_op(sptr, chptr)
 			    && !is_chanprot(sptr, chptr)
 			    && !is_chanowner(sptr, chptr))
