--- sapi/fpm/fpm/fpm_php.c	2010-05-22 19:44:25.000000000 +0200
+++ sapi/fpm/fpm/fpm_php.c	2010-05-22 19:48:07.000000000 +0200
@@ -71,6 +71,43 @@ static void fpm_php_disable(char *value,
 	}
 }
 
+int fpm_php_apply_defines_ex(struct key_value_s *kv, int mode) /* {{{ */
+{
+	TSRMLS_FETCH();
+
+	char *name = kv->key;
+	char *value = kv->value;
+	int name_len = strlen(name);
+	int value_len = strlen(value);
+
+	if (!strcmp(name, "extension") && *value) {
+		zval zv;
+		php_dl(value, MODULE_PERSISTENT, &zv, 1 TSRMLS_CC);
+		return Z_BVAL(zv) ? 1 : -1;
+	}
+
+	if (zend_ini_alter_master(name, name_len+1, value, value_len, /*mode,*/ PHP_INI_STAGE_ACTIVATE TSRMLS_CC) == FAILURE) {
+		return -1;
+	}
+
+	if (!strcmp(name, "disable_functions") && *value) {
+		char *v = strdup(value);
+		PG(disable_functions) = v;
+		fpm_php_disable(v, zend_disable_function TSRMLS_CC);
+		return 1;
+	}
+
+	if (!strcmp(name, "disable_classes") && *value) {
+		char *v = strdup(value);
+		PG(disable_classes) = v;
+		fpm_php_disable(v, zend_disable_class TSRMLS_CC);
+		return 1;
+	}
+
+	return 1;
+}
+/* }}} */
+
 static int fpm_php_apply_defines(struct fpm_worker_pool_s *wp)
 {
 	TSRMLS_FETCH();
--- sapi/fpm/fpm/fpm_php.h	2010-05-22 19:49:58.000000000 +0200
+++ sapi/fpm/fpm/fpm_php.h	2010-05-22 19:50:17.000000000 +0200
@@ -8,6 +8,7 @@
 #include <TSRM.h>

 #include "build-defs.h" /* for PHP_ defines */
+#include "fpm/fpm_conf.h";

 struct fpm_worker_pool_s;

@@ -17,6 +17,7 @@ char *fpm_php_request_method(TSRMLS_D);
 size_t fpm_php_content_length(TSRMLS_D);
 void fpm_php_soft_quit();
 int fpm_php_init_main();
+int fpm_php_apply_defines_ex(struct key_value_s *kv, int mode);
 
 #endif
 
