--- a/lib/err/err.c	2016-10-13 12:27:55.732515855 +0200
+++ b/lib/err/err.c	2016-10-13 12:36:01.173498494 +0200
@@ -143,6 +143,7 @@
    if (!HashTable_Lookup(numTable, (void *) (uintptr_t) errorNumber,
 			 (void **) &info)) {
       char buf[2048];
+      char emptyString[] = { '\0' };
       const char *p;
       size_t n;
 
@@ -156,6 +157,9 @@
       info->number = errorNumber;
       info->string = Util_SafeStrdup(p);
 
+      if (!info->string)
+      	info->string = emptyString;
+
       /*
        * To be safe, make sure the end of the string is at
        * a UTF-8 boundary, but we can only do this when the
