--- a/modules/linux/vmhgfs/module.h	2016-10-13 10:22:07.355785806 +0200
+++ b/modules/linux/vmhgfs/module.h	2016-10-13 10:27:42.733773812 +0200
@@ -59,6 +59,7 @@
 /* The amount of time we'll wait for the backdoor to process our request. */
 #define HGFS_REQUEST_TIMEOUT (30 * HZ)
 
+#define NAME_MAX        255    /* # chars in a file name */
 /*
  * Inode number of the root inode. We set this to be non-zero because,
  * according to glibc source, when the returned inode number in a dirent
--- a/vmhgfs-fuse/module.h	2016-10-13 10:22:09.025785746 +0200
+++ b/vmhgfs-fuse/module.h	2016-10-13 11:16:37.953668840 +0200
@@ -65,7 +65,7 @@
 #include "session.h"
 #include "config.h"
 
-#if defined(__SOLARIS__) || defined(__APPLE__)
+#if !defined(__GLIBC__) || defined(__SOLARIS__) || defined(__APPLE__)
 #define DT_UNKNOWN      0
 #define DT_FIFO         1
 #define DT_CHR          2
