diff -urN php-toolkit-1.0.2.orig/php-select php-toolkit-1.0.2/php-select
--- php-toolkit-1.0.2.orig/php-select	2010-01-17 13:52:25.000000000 +0300
+++ php-toolkit-1.0.2/php-select	2010-01-17 14:00:34.000000000 +0300
@@ -18,7 +18,7 @@
 #
 # where
 #    <component> is one of
-#      php|php-cgi|php-devel|apache|apache2
+#      php|php-cgi|php-fpm|php-devel|apache|apache2
 #
 #    <php-version> is the directory under /usr/lib where PHP is installed
 #
@@ -107,6 +107,7 @@
 	echo "    apache2   - Apache webserver v2.x"
 	echo "    php       - /usr/bin/php (the CLI SAPI)"
 	echo "    php-cgi   - /usr/bin/php-cgi (the CGI SAPI)"
+	echo "    php-fpm   - /usr/bin/php-fpm (the FPM SAPI)"
 	echo "    php-devel - the scripts used to build PECL extensions"
 	echo
 	echo "and where <php-version> is one of:"
diff -urN php-toolkit-1.0.2.orig/php-select-modules/php-fpm.sh php-toolkit-1.0.2/php-select-modules/php-fpm.sh
--- php-toolkit-1.0.2.orig/php-select-modules/php-fpm.sh	1970-01-01 03:00:00.000000000 +0300
+++ php-toolkit-1.0.2/php-select-modules/php-fpm.sh	2010-01-17 13:53:28.000000000 +0300
@@ -0,0 +1,18 @@
+#
+# /usr/share/php-select/php-fpm.sh
+#		Module to manage /usr/bin/php-fpm binary
+#
+#		Written for Gentoo Linux
+#
+# Author	Stuart Herbert
+#		(stuart@gentoo.org)
+#
+# Copyright	(c) 2005 Gentoo Foundation, Inc.
+#		Released under version 2 of the GNU General Public License
+#
+# ========================================================================
+
+G_SYMLINK_SOURCE[0]="bin/php-fpm"
+G_SYMLINK_TARGET[0]="/usr/bin/php-fpm"
+
+. $G_MODULE_PATH/libsymlink.sh
